game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsFactory.java

```diff
@@ -0,0 +1,53 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.DiceRoll;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.NonNull;
+
+/** Build the steps for the fire round (roll dice, select casualties, and mark casualties) */
+@Builder
+public class FireRoundStepsFactory {
+
+  @NonNull final BattleState battleState;
+  @NonNull final BattleActions battleActions;
+  @NonNull final Function<BattleState, List<FiringGroup>> firingGroupSplitter;
+  @NonNull final BattleState.Side side;
+  @NonNull final MustFightBattle.ReturnFire returnFire;
+  @NonNull final BiFunction<IDelegateBridge, RollDice, DiceRoll> diceRoller;
```

Comment:
Side note, we may want to rename 'RollDice'. 'RollDice' and 'diceRoll' being inverses of each other is somewhat confusing, the distinction seems to be very weak, another maintainer might see no real difference between the two terms.

Reply:
Any ideas on renaming it?  `RollDice` is a battle step that sets up the call to `DiceRoll`.  The 2 implementations of `diceRoller` are called `RollAaDice` and `RollNormalDice`.  `RollNormalDice` just calls `DiceRoll` while `RollAaDice` will play a sound and call `DiceRoll`.

One thought is that the other two steps are called `SelectCasualties` and `MarkCasualties` so maybe this should be similar?  Something like `HitCasualties` or `CalculateCasualties`?  Since in LL, dice isn't guaranteed to be rolled.

Reply:
Maybe call it ` RollDiceStep`? or ` RollDiceBattleStep`

> "Since in LL, dice isn't guaranteed to be rolled."

I suppose it's a pedantic argument whether rolling 0 dice is rolling dice. If you took that view, then having a LL roll dice step would still make sense.